### ISSUE TITLE
moment.js is self-deprecating!

### ISSUE THREAD

**Issue by jenbutongit** [XGovFormBuilder/digital-form-builder#115]
**Is your feature request related to a problem? Please describe.**
Time (ha) to use a modern date library.

**Describe the solution you'd like**
Switch to any of these suggested libraries. There is a global ECMA object in proposal stage 2 `Temporal` but it's not safe to use yet. Ideally we use that but for now https://momentjs.com/docs/#/-project-status/

**Describe alternatives you've considered**
we can keep moment ¯\_(ツ)_/¯ 

**Additional context**

**Comment by jenbutongit** [comment]
use date-fns

**Comment by jonnywyatt** [comment]
Timezones are currently used in the 'edit date condition' panel.
Is there a reason this is required, will the person editing the form need / want to specify a timezone when they're writing a date condition? Is there any reason GMT only won't do the job?
![image](https://user-images.githubusercontent.com/471250/102245846-f1d35280-3ef5-11eb-9c1f-5b0891b5e20c.png)

**Comment by jonnywyatt** [comment]
https://github.com/XGovFormBuilder/digital-form-builder/pull/249

**Comment by jenbutongit** [comment]
after #724, `FormModel` and related tests are using moment
